fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/TemplateDruidQuery.java

```diff
@@ -155,9 +155,9 @@ public TemplateDruidQuery nest() {
         LinkedHashSet<Aggregation> innerAggregations = new LinkedHashSet<>();
         LinkedHashSet<Aggregation> outerAggregations = new LinkedHashSet<>();
         for (Aggregation agg : aggregations) {
-            Pair<Aggregation, Aggregation> split = agg.nest();
-            innerAggregations.add(split.getRight());
-            outerAggregations.add(split.getLeft());
+            Pair<Optional<Aggregation>, Optional<Aggregation>> split = agg.nest();
+            innerAggregations.add(split.getRight().get());
+            outerAggregations.add(split.getLeft().get());
```

Comment:
Nope.
You need to do something like:
split.getLeft().stream().each(innerAggregations::add)
or
if( ! split.getLeft().isEmpty() ) { innerAggregations.add(split.getLeft()) }